### PR TITLE
project-18-shaun-newmark

### PR DESCRIPTION
**Pull request by newmarkdev**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,37 +1,159 @@
...
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import btc from './data/btc.json'
```
Comment (Marielle8) [inline server.js:8]: Nice that you went for another database then the ones that were suggested 👍

File: server.js [hunk 2]
```diff
@@ -1,37 +1,159 @@
+/* eslint-disable max-len */
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+// import dotenv from 'dotenv'
```
Comment (Marielle8) [inline server.js:5]: Remove this since you don't use it?

File: server.js [hunk 3]
```diff
@@ -1,37 +1,159 @@
...
+  Name: String,
+  Symbol: String,
+  Date: String,
+  High: Number,
+  Low: Number,
+  Open: Number,
+  Close: Number,
+  Volume: Number,
+  Marketcap: Number
+});
```
Comment (Marielle8) [inline server.js:27]: Nice!

File: server.js [hunk 4]
```diff
@@ -1,37 +1,159 @@
...
+app.use(express.json())
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', (_, res) => {
+  res.send(listEndpoints(app));
 })
 
+app.get('/btc-historical-prices', async (req, res) => {
```
Comment (Marielle8) [inline server.js:55]: Maybe add a comment here on what this endpoint do just to be consistent with the comments=)

File: server.js [hunk 5]
```diff
@@ -1,37 +1,159 @@
...
+  }
+});
+// 2021
+app.get('/btc-historical-prices/2021', async (req, res) => {
+  const btcData = await BtcHistorical.find();
+  const btc2021 = btcData.filter((item) => item.SNo === 2859)
+  if (btc2021) {
+    res.status(200).json({ dataFeb2021: { closingPrice: btc2021[0].Close, marketCap: btc2021[0].Marketcap } });
+  } else {
+    res.status(404).json({ error: 'Not found' });
```
Comment (Marielle8) [inline server.js:153]: Nice to add error status

File: server.js [hunk 6]
```diff
@@ -1,37 +1,159 @@
...
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import btc from './data/btc.json'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+// dotenv.config()
```
Comment (Marielle8) [inline server.js:10]: Remove this since you don't use it?

**Review by Marielle8 — APPROVED** [review]
Nice work Shaun! As far as I can see you have fulfilled all the blue requirments for week 18! Way to go 🙌😃🕺

File: README.md [hunk 7]
```diff
@@ -1,13 +1,11 @@
...
 
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+The solution was to use mongoDB and mongoose, practicing mongoose methods and seeding the cloud database with the data. I used a sample data set of historical prices for bitcoin from 2013 to show prices and market cap in each year until 2021. 
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+[Bitcoin Historical Prices](https://btc-historicals.herokuapp.com/)
```
Comment (KarolCislowski) [inline README.md:11]: short documentation would be nice 🤓

File: server.js [hunk 8]
```diff
@@ -1,37 +1,159 @@
...
+// 2015
+app.get('/btc-historical-prices/2015', async (req, res) => {
+  const btcData = await BtcHistorical.find();
+  const btc2015 = btcData.filter((item) => item.SNo === 733)
+  if (btc2015) {
+    res.status(200).json({ dataMay2015: { closingPrice: btc2015[0].Close, marketCap: btc2015[0].Marketcap } });
+  } else {
+    res.status(404).json({ error: 'Not found' });
+  }
+});
```
Comment (KarolCislowski) [inline server.js:95]: they all are the same, please try to refactor them 🤓

**Review by KarolCislowski — APPROVED** [review]
Nice API 👍🏻

**Pull request closed without merging**